src/hooks/useImmediateEffect.test.ts

```diff
@@ -8,14 +8,37 @@ it('calls effects immediately on mount', () => {
   const effect = jest.fn();
 
   spy.mockImplementation(useEffect);
-  renderHook(() => useImmediateEffect(effect, [effect]));
 
-  expect(effect).toHaveBeenCalledTimes(1);
-  expect(effect).toHaveBeenCalledTimes(1);
+  renderHook(() => {
+    useImmediateEffect(effect, [effect]);
+    expect(effect).toHaveBeenCalledTimes(1);
+  });
 
+  expect(effect).toHaveBeenCalledTimes(1);
```

Comment:
Not sure I totally get the logic behind the second `.toHaveBeenCalledTimes` here. As I read this, I'd expect `useImmediateEffect` to get called once on first render (which we verify with the first assertion), then I wouldn't expect us to need to assert again. So can you explain the reasoning behind having this second assertion?

Reply:
react-hooks-testing-library had this weird thing where it'd execute effects after the initial mount automatically.

So if the initial effect was actually executed after manual call inside the initial mount, we'd see two calls here